libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_ptx_compiles.pass.cpp

```diff
@@ -52,14 +57,15 @@ __global__ void test_bulk_tensor(CUtensorMap* map)
 __global__ void test_bulk(void* gmem)
 {
   __shared__ int smem;
-  __shared__ barrier* bar;
+  __shared__ char barrier_data[sizeof(barrier)];
+  barrier& bar = *reinterpret_cast<barrier*>(&barrier_data);
```

Comment:
Q: Why do we need a `reinterpret_cast` here and not a `bit_cast` like above?

Reply:
I want to be 100% sure that this is indeed pointing to the shared memory and not doing some internal copies